### DISCUSSION TITLE
DIAMOND v2.0.11

### DISCUSSION THREAD

**Discussion by bbuchfink** [bbuchfink/diamond#479] [General]
- Fixed a bug that could cause invalid output when using `--masking 0` combined with the global ranking mode.
- Enabled lazy repeat masking in the query-indexed and contiguous seed modes when using global ranking.
- Added detection of cache size to auto-enable query-indexed mode.

<hr /><em>This discussion was created from the release <a href='https://github.com/bbuchfink/diamond/releases/tag/v2.0.11'>DIAMOND v2.0.11</a>.</em>

**Comment by FredericBGA** [comment]
Hello

I've compiled this version (as well as the 2.0.10) from sources.
The `prepdb` command is not included?

> #diamond version
> diamond version 2.0.11
> #diamond prepdb -d Tmp/Rice.fasta
> Error: Invalid command: prepdb. To print help message: diamond help

Do I need to add something during the compilation process?

**Comment by bbuchfink** [comment]
`prepdb` is only for using BLAST databases, so you need to compile with BLAST database support if you want to use that, which is pretty elaborate (see installation in the wiki).

**Reply by FredericBGA** [reply]
sorry, I've missed the wiki dedicated to this specific installation.

I've compiled Blast again with the recommended options.

> [root@universe build#]ls /softs/biotools/diamond-blast-2.12.0/lib/libseqdb
> libseqdb-static.a  libseqdb.a

But the libs are not found, I will try to understand why now.

> [root@universe build#]cmake -DBLAST_INCLUDE_DIR=/softs/biotools/diamond-blast-2.12.0/include/ncbi-tools++  -DBLAST_LIBRARY_DIR==/softs/biotools/diamond-blast-2.12.0/lib ..
> -- The C compiler identification is GNU 6.2.0
> -- The CXX compiler identification is GNU 6.2.0
> -- Check for working C compiler: /softs/gcc/6.2.0/bin/gcc
> -- Check for working C compiler: /softs/gcc/6.2.0/bin/gcc -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Check for working CXX compiler: /softs/gcc/6.2.0/bin/c++
> -- Check for working CXX compiler: /softs/gcc/6.2.0/bin/c++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Performing Test HAS_GNUPP14
> -- Performing Test HAS_GNUPP14 - Success
> -- Found ZLIB: /usr/lib64/libz.so (found version "1.2.3")
> -- Looking for include file pthread.h
> -- Looking for include file pthread.h - found
> -- Looking for pthread_create
> -- Looking for pthread_create - not found
> -- Looking for pthread_create in pthreads
> -- Looking for pthread_create in pthreads - not found
> -- Looking for pthread_create in pthread
> -- Looking for pthread_create in pthread - found
> -- Found Threads: TRUE
> BLAST_INCLUDE_DIR: /softs/biotools/diamond-blast-2.12.0/include/ncbi-tools++
> BLAST_LIBRARY_DIR: =/softs/biotools/diamond-blast-2.12.0/lib
> CMake Error at CMakeLists.txt:297 (message):
>   Unable to find BLAST library: seqdb

**Reply by bbuchfink** [reply]
Don't use two = here: `-DBLAST_LIBRARY_DIR==`

**Reply by FredericBGA** [reply]
rhoo, shame on me! thank you...
